nilearn/connectome/group_sparse_cov.py

```diff
@@ -670,7 +670,7 @@ def fit(self, subjects, y=None):
                 ensure_min_samples=2,
             )
 
-        self.n_features_in_ = next(iter({s.shape[1] for s in subjects}))
+        self.n_features_in_ = next(iter(s.shape[1] for s in subjects))
```

Comment:
Can I be annoying and ask you to also move the line after the empirical covariance is computed?
Reason: all subjects should have the same number of features but this is only checked by `empirical_covariances` so `self.n_features_in_` should be set after that.

Don't hesitate to add a comment to explain it.

I added this line a couple of days ago and did not realize that this bit would "read strange".